src/rover/main.cpp

```diff
@@ -0,0 +1,174 @@
+	std::cout << "Camera 0: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 1: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 2: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 3: " << camera::get_error_string(err0) << std::endl;
+#endif
+	// Start cameras
+	err0 = camera::start(&session0);
+	err1 = camera::start(&session1);
+	err2 = camera::start(&session2);
+	err3 = camera::start(&session3);
+#ifdef DEBUG
+	std::cout << "Camera 0: " << camera::get_error_string(err0) << std::endl;
+	std::cout << "Camera 1: " << camera::get_error_string(err1) << std::endl;
+	std::cout << "Camera 2: " << camera::get_error_string(err2) << std::endl;
+	std::cout << "Camera 3: " << camera::get_error_string(err3) << std::endl;
+#endif
+	// Open UDP connection
+	network::Connection conn;
+	network::Error net_err = network::connect(&conn, "192.168.1.1", 45545, 45545);
+	if((net_err != network::Error::OK)) {
+		std::cerr << "[!]Failed to connect to rover!" << std::endl;
+	}
+
+	/*
+	 * MAIN LOOP
+	 * 1. Process Info Locally
+	 * 2. Send out packets
+	 */
+	while(true) {
+		/*
+		 * 1. Process Info Locally
+		 * - Grabbing frames
+		 */
+#ifdef TIME
+	auto start_all = high_resolution_clock::now();
+#endif
+		//Grab a frame
+		uint8_t* frame_buffer;
+		std::size_t frame_size;
+		err1 = camera::grab_frame(&session0, &frame_buffer, &frame_size);
+		frame_counter ++;
+#ifdef DEBUG
+		// Print camera buffer info
+		std::cout << (void*)frame_buffer << std::endl;
+		std::cout << "1" << camera::get_error_string(err1) << std::endl;
+#endif
+#ifdef TIME
+	auto end_grab_frame = high_resolution_clock::now();
+	std::cout << 
+		"[LOCAL_TIME] Time to GRAB FRAME from session0: " <<
+		duration_cast<milliseconds>(end_grab_frame - start_all).count() <<
+		"ms" <<
+	std::endl;
+#endif
+
+
+
+		/*
+		 * 2. Send out packets
+		 * - Send Logging packets
+		 * - Create camera messages and send Camera packets
+		 */
+		network::poll_incoming(&conn);
+		network::Message message;
+		// Receive incoming messages //
+		while(network::dequeue_incoming(&conn, &message)) {
+				switch(message.type) {
+					case network::MessageType::HEARTBEAT:
+					{
+						network::Buffer* outgoing = network::get_outgoing_buffer();
+						network::queue_outgoing(&conn, network::MessageType::HEARTBEAT, outgoing);
+						break;
+					}
+					default:
+						break;
+				}
+		}
+		// THIS IS SAMPLE CODE to send outgoing packets //
+		// Log messages
+		network::Buffer* log_buffer = network::get_outgoing_buffer();
+		char s[] = "[LOG] Hewwo Wowld!";
+		network::LogMessage log_message = {static_cast<uint8_t>(strlen(s)), s};
+		network::serialize(log_buffer, &log_message);
+
+		network::queue_outgoing(&conn, network::MessageType::LOG, log_buffer);
+
+		// Camera Frames
+		//Temporary 'for' loop to use the same frame from session0 for all streams
+		for (unsigned int i = 0; i < 4; i++) {
+			uint8_t num_buffers = (frame_size/CAMERA_MESSAGE_SIZE) + 1;
```

Comment:
We're aware this is integer division. What happens if, for some reason, the numbers aren't multiples of each other? How can we better handle that

Reply:
I've added a comment to highlight this